File: legacy/OrderMatchingEngine/src/main/java/com/ndgndg91/ordermatchingengine/order/dto/request/ModifyOrderRequest.java
package com.ndgndg91.ordermatchingengine.order.dto.request;

import com.ndgndg91.ordermatchingengine.order.OrderType;
import com.ndgndg91.ordermatchingengine.order.PriceType;
import com.ndgndg91.ordermatchingengine.order.Symbol;
import com.ndgndg91.ordermatchingengine.order.validation.ValueOfEnum;
import lombok.Getter;
import lombok.ToString;

import javax.validation.constraints.*;
import java.math.BigDecimal;

@Getter
@ToString
public final class ModifyOrderRequest {
    @NotNull(message = "order id is required")
    private String orderId;
    @ValueOfEnum(enumClass = OrderType.class)
    @NotNull(message = "order type is required")
    private String orderType;
    @ValueOfEnum(enumClass = Symbol.class)
    @NotNull(message = "symbol is required")
    private String symbol;
    @Min(1)
    @Max(Integer.MAX_VALUE)
    @NotNull(message = "shares is required")
    private Integer shares;
    @ValueOfEnum(enumClass = PriceType.class)
    @NotNull(message = "price type is required")
    private String priceType;
    @DecimalMin("0.0")
    @DecimalMax("9999999999.99")
    @Digits(integer = 10, fraction = 2)
    private BigDecimal price;
}

File: legacy/OrderMatchingEngineKotlin/src/main/kotlin/com/ndgndg91/ordermatchingenginekotlin/order/dto/request/ModifyOrderRequest.kt
package com.ndgndg91.ordermatchingenginekotlin.order.dto.request

import com.ndgndg91.ordermatchingenginekotlin.order.validation.OrderTypeValue
import com.ndgndg91.ordermatchingenginekotlin.order.validation.PriceTypeValue
import com.ndgndg91.ordermatchingenginekotlin.order.validation.SymbolValue
import java.math.BigDecimal
import javax.validation.constraints.*

data class ModifyOrderRequest(
    @field:NotNull(message = "order id is required") val orderId: String?,
    @field:OrderTypeValue @field:NotNull(message = "order type is required") val orderType: String?,
    @field:SymbolValue @field:NotNull(message = "symbol is required") val symbol: String?,
    @field:Min(1) @field:Max(Int.MAX_VALUE.toLong()) @field:NotNull(message = "shares is required") val shares: Int?,
    @field:PriceTypeValue @field:NotNull(message = "price type is required") val priceType: String?,
    @field:DecimalMin("0.0") @field:DecimalMax("9999999999.99") @field:Digits(integer = 10, fraction = 2)
    @field:NotNull(message = "price is required") val price: BigDecimal?
)